src/pedal/config.rs

```diff
@@ -0,0 +1,69 @@
+use crate::{pedal, Pedal as PedalTrait, Result};
+use cpal::StreamConfig;
+use serde::Deserialize;
+use Pedal::*;
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    pedals: Vec<Pedal>,
+}
+
+impl Config {
+    pub fn from(yaml: &str) -> Result<Config> {
+        Ok(serde_yaml::from_str(yaml)?)
+    }
+
+    pub fn default() -> Self {
+        Self {
+            pedals: vec![Transparent],
+        }
+    }
+
+    pub fn to_pedals(&self, stream_config: &StreamConfig) -> Result<Vec<pedal::Boxed>> {
+        self.pedals
+            .iter()
+            .map(|pedal| pedal.to_pedal(stream_config))
+            .collect()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type")]
+enum Pedal {
+    Transparent,
+    Delay(DelayConfig),
+}
+
+impl Pedal {
+    fn to_pedal(&self, stream_config: &StreamConfig) -> Result<pedal::Boxed> {
+        Ok(match self {
+            Transparent => pedal::Transparent::new().boxed(),
+            Delay(delay_config) => delay_config.to_pedal(stream_config)?,
+        })
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct DelayConfig {
+    pub level: f32,
+    pub delay_ms: u32,
+    pub num: u32,
+}
+
+impl DelayConfig {
+    fn to_pedal(&self, stream_config: &StreamConfig) -> Result<pedal::Boxed> {
+        let mut pedals = (1..=self.num)
+            .map(|n| {
+                Ok(
+                    pedal::Delay::new(stream_config, self.level.powi(n as i32), self.delay_ms * n)?
```

Comment:
What's the math here with `.powi()`?

Reply:
It's exponentially decreasing the level with the number of the delay "unit". So if the config sets the level to 0.5, you get:
- 0.5^1
- 0.5^2
- 0.5^3
- etc.